src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import React from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { todos } from './helpers/getTodos';
+import users from './api/users';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { getUser } from './helpers/getUser';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
-
-export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
-    </div>
-  );
-};
+type State = Todo;
+
+export class App extends React.Component<{}, State> {
+  state = {
+    id: Math.max(...todos.map(todo => todo.id)) + 1,
+    title: '',
+    userId: 0,
+    completed: false,
+  };
+
+  inputStatusMonitoring = false; // this property used in order to comply with this requirement: "errors should appear only after clicking the `Add` button"
```

Comment:
move it into state

Reply:
In general as I understand I must avoid using such kind of properties? Or is there any case they are more preferable than state properties?